src/test/java/org/kohsuke/github/GHContentIntegrationTest.java

```diff
@@ -170,4 +171,25 @@ public void testGetFileContentWithNonAsciiPath() throws Exception {
         final GHContent fileContent2 = repo.getFileContent(fileContent.getPath());
         assertThat(IOUtils.readLines(fileContent2.read(), StandardCharsets.UTF_8), hasItems("test"));
     }
+
+    @Test
+    public void testGetFileContentWithSymlink() throws Exception {
+        final GHRepository repo = gitHub.getRepository("hub4j-test-org/GHContentIntegrationTest");
+
+        final GHContent fileContent = repo.getFileContent("ghcontent-ro/a-symlink-to-a-file");
+        // for whatever reason GH says this is a file :-o
+        assertThat(IOUtils.toString(fileContent.read(), StandardCharsets.UTF_8), is("thanks for reading me\n"));
+
+        final GHContent dirContent = repo.getFileContent("ghcontent-ro/a-symlink-to-a-dir");
+        // but symlinks to directories are symlinks!
+        assertThat(dirContent, allOf(hasProperty("target", is("a-dir-with-3-entries")),
+                                     hasProperty("type", is("symlink"))));
+        
+        // TODO somehow
+        /*
+        final GHContent fileContent2 = repo.getFileContent("ghcontent-ro/a-symlink-to-a-dir/entry-one");
+        // this needs special handling and will 404 from GitHub
+        assertThat(IOUtils.toString(fileContent.read(), StandardCharsets.UTF_8), is(""));
+        */
```

Comment:
There are ways to do this but they are inefficient/costly.

Reply:
agreed,  the API does not seem to support this, so given this is a mapping of the API to java I am not sure we should either.

That said, I was looking for this in order to solve https://issues.jenkins-ci.org/browse/JENKINS-62922 and if someone wants to acheive this I am not sure that making all users of the API handle it themselves is the correct approach.